lib/scripts.js

```diff
@@ -303,59 +302,28 @@ var scripts = {
    * Takes a lock
    */
   takeLock: function(queue, job, token, renew){
-    console.log('attempting lock for job id #', job.jobId)
-    var lockCall;
-    if (renew){
-      lockCall = 'redis.call("SET", KEYS[1], ARGV[1], "PX", ARGV[2])';
-    } else {
-      lockCall = 'redis.call("SET", KEYS[1], ARGV[1], "PX", ARGV[2], "NX")';
-    }
-
-    var script = [
-      'if(' + lockCall + ') then',
-      // Mark the job as having been locked at least once. Used to determine if the job was stalled.
-      ' redis.call("HSET", KEYS[2], "lockAcquired", "1")',
-      ' return 1',
-      'else',
-      ' return 0',
-      'end'
-    ].join('\n');
-
-    var args = [
-      queue.client,
-      'takeLock' + (renew ? 'Renew' : ''),
-      script,
-      2,
-      job.lockKey(),
-      queue.toKey(job.jobId),
-      token,
-      queue.LOCK_RENEW_TIME
-    ];
 
-    return execScript.apply(scripts, args);
+    var lock = job.lock;
+    if (renew && !lock) {
+      throw new Error('Unable to renew nonexisting lock');
+    } else if (renew) {
+      return lock.extend(queue.LOCK_RENEW_TIME);
+    } else if (lock) {
```

Comment:
I think the if here is not needed, lock must be defined if we reached this point.